parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -212,8 +212,20 @@ private static MetadataFilter filter(boolean skipRowGroups) {
   }
 
   private static <T> List<T> runAllInParallel(int parallelism, List<Callable<T>> toRun) throws ExecutionException {
-    LOG.info("Initiating action with parallelism: {}", parallelism);
-    ExecutorService threadPool = Executors.newFixedThreadPool(parallelism);
+    if (toRun.isEmpty()) {
```

Comment:
I think this is a reasonable and safe optimization. If there is nothing to run, there is no need to create a threadpool.